File: 剑指Offer/调整数组顺序使奇数在偶数之前/changeArray.py
# -*- coding:utf-8 -*-
class Solution:
    def reOrderArray(self, array):
        # write code here
        left,right = len(array)-2,len(array)-1
        while left>=0:
            if array[left] %2 == 0 and array[right] %2 != 0 and right-left == 1:
                array[left],array[right] = array[right],array[left]
                left -= 1
                right -= 1
            elif array[left] %2 == 0 and array[right] %2 != 0 and right-left != 1:
                i = left
                temp = array[i]
                while i < right:
                    array[i] = array[i+1]
                    i+=1
                array[right] = temp
                left -= 1
                right -= 1
            elif array[left]%2 != 0 and array[right]%2!=0:
                left-=1
            else:
                left -= 1
                right -= 1
        return array

File: 剑指Offer/调整数组顺序使奇数在偶数之前/changeArray.java
public class Solution {
    public void reOrderArray(int [] array) {
        int left = array.length - 2;
        int right = array.length - 1;
        while(left>-1){
            if(array[left]%2 == 0 && array[right]%2 != 0 && right - left == 1){
                int temp = array[left];
                array[left] = array[right];
                array[right] = temp;
                left--;
                right--;
            }
            else if(array[left]%2 == 0 && array[right]%2 != 0 && right - left > 1){
                int i = left;
                int temp = array[left];
                while(i < right){
                    array[i] = array[++i];
                }
                array[right] = temp;
                left--;
                right--;
            }
            else if(array[left]%2 != 0 && array[right]%2 != 0){
                left--;
            }
            else{
                left--;
                right--;
            }
        }
    }
}